pkg/rotator/rotator.go

```diff
@@ -124,7 +167,14 @@ type CertRotator struct {
 }
 
 // Start starts the CertRotator runnable to rotate certs and ensure the certs are ready.
-func (cr *CertRotator) Start(stop <-chan (struct{})) error {
+func (cr *CertRotator) Start(stop <-chan struct{}) error {
+	if cr.reader == nil {
+		return errors.New("nil reader")
+	}
+	if !cr.reader.WaitForCacheSync(stop) {
+		return errors.New("failed waiting for reader to sync")
```

Comment:
If this sync fails at start up, how does it recover?

Reply:
This call blocks until the caches sync, and only fails if the stop channel is closed prior to that happening. If that happened, a shutdown was requested and we don't need to retry.
Transient failures are managed and retired within the informer cache.